### DISCUSSION TITLE
How to remove deprecated entities from devices

### DISCUSSION THREAD

**Discussion by legantois** [make-all/tuya-local#1961] [General]
i am making reference to this closed point #1842 

I get warning that switch entities of one device are deprecated and valve should be used
2024-05-28 11:24:32.093 WARNING (MainThread) [custom_components.tuya_local.helpers.config] The use of switch for Dual water timer is deprecated and should be replaced by valve.
2024-05-28 11:24:32.093 WARNING (MainThread) [custom_components.tuya_local.helpers.config] The use of switch for Dual water timer is deprecated and should be replaced by valve.

I tried as suggested
I updated all automations, actions...

I disabled the switch entities -> still got the same warning

I removed the device and reinstalled it, disabled entities -> did not work

I still have as warning at each reboot:

could it be possible to modify the configuration of this device so that when i reinstall it the switch entity is not created anymore ?

or anything else that could remove this warning ?

Many thanks for your help

**Comment by make-all** [comment]
The warning can be ignored if you are not making use of the deprecated entity.  There may be a better place to output the warning, so that it is only output when the deprecated entities are enabled.  In the current release, it is always output if you are using a device which has deprecated entities.

**Comment by legantois** [comment]
Hello Jason,

Thanks for your answer. Indeed two warnings a reboot is not a big issue

Are there any plans / process to remove these deprecated entities from the device config in your integration ? or a way i could do by myself ?

**Reply by make-all** [reply]
The last batch of deprecated entities I removed was after around 6 months. The valve entity was added in March, so probably still a bit early for final removal of the old switch entities they replaced.
